provisioner/data.js

```diff
@@ -86,6 +87,107 @@ var WorkerType = base.Entity.configure({
   context: ['provisionerId', 'keyPrefix'],
 });
 
+// We want to add a lastModified field
+WorkerType = WorkerType.configure({
+  version: 2,
+  properties: {
+    // These fields are documented in Version 1 of this Entity
+    workerType: base.Entity.types.String,
+    minCapacity: base.Entity.types.Number,
+    maxCapacity: base.Entity.types.Number,
+    scalingRatio: base.Entity.types.Number,
+    minPrice: base.Entity.types.Number,
+    maxPrice: base.Entity.types.Number,
+    canUseOndemand: base.Entity.types.JSON, // delete this
+    canUseSpot: base.Entity.types.JSON, // delete this
+    instanceTypes: base.Entity.types.JSON,
+    regions: base.Entity.types.JSON,
+    // Store the date of last modification for this entity
+    lastModified: base.Entity.types.Date,
+    // Global base UserData object for overwriting
+    userData: base.Entity.types.JSON,
+    // Global base LaunchSpecification object for overwriting
+    launchSpec: base.Entity.types.JSON,
+    // Global base secrets object for overwriting
+    secrets: base.Entity.types.JSON,
+    // Global base scope list for appending, these are the scopes that
+    // temporary TC credentials will be issued against
+    scopes: base.Entity.types.JSON,
+  },
+  migrate: function (item) {
+    console.log('Upgrading a workerType to version 2');
+    console.log(JSON.stringify(item, null, 2));
+    var oldUserData;
+    if (item.launchSpecification.UserData) {
+      oldUserData = JSON.parse(new Buffer(item.launchSpecification.UserData, 'base64').toString());
+    } else {
+      oldUserData = {};
+    }
+    var oldLaunchSpec = item.launchSpecification;
+    delete item.launchSpecification;
```

Comment:
I haven't tested modifying the existing object... It should just be a flat object, and I think it does work.
But it's not part of any of the examples or tests from `taskcluster-base`, [see docs](https://github.com/taskcluster/taskcluster-base/blob/26cd5e3b824f52016440512ccea5f954b774f26c/lib/entity.js#L211-L216).

If you want to use this pattern, please add a test doing it to `taskcluster-base`, or just make a fresh object, I think it's safer...